### PR TITLE
Validate countries states_required attribute when states present

### PR DESCRIPTION
**Pull request by tanmay3011**

### CODE REVIEW

**Comment by jitendra** [comment]
@tanmay3011 lets discuss this today

File: core/app/models/spree/country.rb [hunk 1]
```diff
@@ -14,6 +14,7 @@ class Country < Spree::Base
     before_destroy :ensure_not_default
 
     validates :name, :iso_name, presence: true
+    validate :ensure_states_required, if: -> { states.present? }
```
Comment (jitendra) [inline core/app/models/spree/country.rb]: change the name to can_have_states?

File: core/spec/models/spree/country_spec.rb [hunk 2]
```diff
@@ -12,6 +13,30 @@
...
+        subject { america.valid? }
+        it { expect(subject).to be_falsy }
+
+        context 'states_required should be checked error on country' do
+          before { subject }
+          it { expect(america.errors[:states_required]).to include(Spree.t(:states_required_invalid)) }
+        end
+      end
+
+      context 'associated states not present' do
```
Comment (jitendra) [inline core/spec/models/spree/country_spec.rb]: does not have states

File: core/spec/models/spree/country_spec.rb [hunk 3]
```diff
@@ -12,6 +13,30 @@
...
+      context 'associated states present' do
+        before do
+          america.states << state
+          america.states_required = false
+        end
+
+        subject { america.valid? }
+        it { expect(subject).to be_falsy }
+
+        context 'states_required should be checked error on country' do
```
Comment (jitendra) [inline core/spec/models/spree/country_spec.rb]: should be checked for error on country

File: core/config/locales/en.yml [hunk 4]
```diff
@@ -1293,6 +1293,7 @@ en:
       shipped: Shipped
       void: Void
     states_required: States Required
+    states_required_invalid: should be checked, country has associated states.
```
Comment (jitendra) [inline core/config/locales/en.yml]: State Required is set to not allow states for this country.

**Comment by tanmay3011** [comment]
Done!

**Comment by jitendra** [comment]
:+1: looks good

**Pull request closed without merging**